internal/rbd/controllerserver.go

```diff
@@ -342,8 +353,13 @@ func flattenParentImage(ctx context.Context, rbdVol *rbdVolume, cr *util.Credent
 // repairExistingVolume checks the existing volume or snapshot and makes sure
 // that the state is corrected to what was requested. It is needed to call this
 // when the process of creating a volume was interrupted.
-func (cs *ControllerServer) repairExistingVolume(ctx context.Context, req *csi.CreateVolumeRequest,
-	cr *util.Credentials, rbdVol *rbdVolume, rbdSnap *rbdSnapshot) (*csi.CreateVolumeResponse, error) {
+func (cs *ControllerServer) repairExistingVolume(
+	ctx context.Context,
+	req *csi.CreateVolumeRequest,
+	cr *util.Credentials,
+	rbdVol *rbdVolume,
+	rbdSnap *rbdSnapshot,
+	setMetadata bool) (*csi.CreateVolumeResponse, error) {
```

Comment:
you dont need to pass metMetadata use cs.SetMetadata from the receiver? instead of request just send parameters as input to this function?